### PR TITLE
Added Karatsuba Multiplication in C++

### PR DESCRIPTION
**Pull request by Aftaab99**
**Fixes issue:**

**Changes:**
Added Karatsuba multiplication in C++

### CODE REVIEW

File: code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#include<iostream>
+using namespace std;
```
Comment (arnavb) [inline code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp]: Please don't use this.

File: code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
+#include<iostream>
```
Comment (arnavb) [inline code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp]: Add a space before `<`

File: code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+	if(strLen1>strLen2){
+		int numZerosToPad = strLen1 - strLen2;
+		for(int i=0; i< numZerosToPad; i++)
+			bitStr2 = '0' + bitStr2;
+	}
+
+	else if(strLen2>strLen1){
+		int numZerosToPad = strLen2 - strLen1;
+		for(int i=0; i< numZerosToPad; i++)
```
Comment (arnavb) [inline code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp]: Please spaces after/before all operators. Please spaces after keywords like `if`, `for`, etc.

File: code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+{
+    string result="";
+    int len = equalLength(bitStr1, bitStr2);
+
+    // For first bit addition carry is 0
+    int carry = 0;
+    for(int i=len-1; i>=0; i--){
+		// We need to convert '0' or '1' to 0 or 1. Subtracting '0' from the character, subtracts
+		// their ascii values which results in 0 or 1.
+		int bit1 = bitStr1.at(i) - '0';
```
Comment (arnavb) [inline code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp]: Just index the string unless you plan on doing exception handling.

File: code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+string addBitStrings(string bitStr1, string bitStr2)
+{
+    string result="";
+    int len = equalLength(bitStr1, bitStr2);
+
+    // For first bit addition carry is 0
+    int carry = 0;
+    for(int i=len-1; i>=0; i--){
```
Comment (arnavb) [inline code/mathematical_algorithms/src/karatsuba_multiplication/karatsuba_multiplication.cpp]: Pre-decrement.

**Comment by arnavb** [comment]
@Aftaab99 Why did you close your PR? You can make further commits to apply the changes I requested for this PR.

**Comment by Aftaab99** [comment]
@arnavb I accidentally deleted the forked repository I had used to create the pull request. I was no longer able to edit it. I've made the changes you've asked me and created a new pull request. 
Sorry for incompetency, I'm very new to open source.

**Comment by arnavb** [comment]
Don't worry about it! Just don't delete your fork next time unless your PR is merged/closed.

**Pull request closed without merging**